### DISCUSSION TITLE
Help making a function out of this simple query

### DISCUSSION THREAD

**Discussion by ashbomb328** [supabase/supabase#10390] [General]
I need help making this into a function and calling it. It uses the table called "Netflix" and "netflix_credits" It joins them based on one attribute. I wanted to make a function that could be easily used for our website and I was having trouble with a way to test it.

create or replace function get_netflix ()
  returns setof netflix
  language plpgsql
  as
  $$
    begin
    return query
  SELECT Distinct title, description, imdb_score, release_year, genres,production_countries, type, seasons, age_certification, role, name
    FROM netflix, netflix_credits
    where netflix_credits_id = netflix_id and role = 'DIRECTOR';
    end;
  $$

The above code is what I did to make the function inside of superbase sql editor.

SELECT * from get_netflix()

The above code is what i tried testing the function with. It does not return results, so does anyone know how I could write a test case and or sql code that correctly calls this function?

**Comment by steve-chavez** [comment]
> so does anyone know how I could write a test case and or sql code that correctly calls this function?

Check this guide about pgTAP tests https://supabase.com/docs/guides/database/testing
